openarc/openarc-ar.c

```diff
@@ -501,6 +501,9 @@ ares_parse(u_char *hdr, struct authres *ar)
 				if (n > 0)
 					n--;
 
+				prevstate = state;
+				state = 2;
```

Comment:
State 2 is not terminal, which I believe means "none; dkim=pass ..." would be accepted although it's not valid.  You should also add it to the list of fatal states at the end of this function for this to be thorough.